dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+CLANG_FORMAT=${CLANG_FORMAT:="clang-format"}
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "the current setting is "
+    echo "CURRENT_DIR ${CURRENT_DIR}"
+    echo "SCRIPT_DIR ${SCRIPT_DIR}"
+    echo "ROOT_DIR ${ROOT_DIR}"
+    echo "ROOT_BUILD_DIR ${ROOT_BUILD_DIR}"
+    echo "GTEST_HEADER_DIR ${GTEST_HEADER_DIR}"
+    echo "CUDA_HEADER_DIR ${CUDA_HEADER_DIR}"
+    echo "ROOT_BUILD_DIR ${ROOT_BUILD_DIR}"
+    echo "CLANG_FORMAT ${CLANG_FORMAT}"
+fi
+if [[ "${CUDA_HEADER_DIR}" == "" ]]; then
+    echo "Please set the CUDA_HEADER_DIR in environment"
+    exit 1
+fi
+# move to working_directory
+cd working_directory
+
+
+KERNEL_SYNTAX_START="<<<"
+KERNEL_SYNTAX_END=">>>"
+DEVICE_CODE_SYNTAX="#include \"(common.*)\""
+FUNCTION_END=");"
+SPECIAL_SUFFIX="_AUTOHOSTFUNC"
+EXTRACT_KERNEL="false"
+GLOBAL_FILE="global_kernel"
+CONFIG_SELECTION_SUFFIX="_CONFIG"
+
+check_closed() {
+    local str="$1"
+    str="${str//->}"
+    str_start="${str//[^(<\[]}"
+    str_end="${str//[^>)\]]}"
+    if [[ "${#str_start}" -eq "${#str_end}" ]]; then
+        echo "true"
+    else
+        echo "false"
+    fi
+}
+
+convert_syntax() {
+    local syntax_regex="([^<>]*)(<[^<>]*>)?<<<(.*)>>>(.*)"
+    local str="$1"
+    str=$(echo "${str}" | sed -E 's/ +/ /g')
+    # echo "str ${str}"
+    local temp=""
+    local num=0
+    local var=""
+    if [[ "${str}" =~ ${syntax_regex} ]]; then
+        content="${BASH_REMATCH[3]}"
+        # echo "// content ${content}"
+        IFS=',' read -ra par <<< "$content"
+        for variable in "${par[@]}"; do
+            if [ -n "${temp}" ]; then
+                temp="${temp},"
+            fi
+            temp="${temp}${variable}"
+            # echo "// temp ${temp}"
+            is_closed=$(check_closed "$temp")
+            if [[ "${is_closed}" = "true" ]]; then
+                # echo "// temp ${temp}"
+                num=$((num+1))
+                # the last stream is zero, use the queue from exec
+                if [[ "${num}" -eq 4 ]] && [[ "${temp// }" -eq 0 ]]; then
+                    num=$((num-1))
+                else
+                    if [ -n "${var}" ]; then
+                        var="${var},"
+                    fi
+                    var="${var}${temp}"
+                fi
+                temp=""
+            fi
+        done
+        # echo "// var ${var}"
+        # echo "var ${var}"
+        if [[ "${num}" -lt 2 ]]; then
+            var="Error"
+        else
+            if [[ ${num} -eq 2 ]]; then
+                var="${var}, 0"
+                num=$((num+1))
+            fi
+            if [[ ${num} -eq 3 ]]; then
+                # var="${var}, exec->get_queue()"
+                var="${var}, GET_QUEUE"
+                num=$((num+1))
+            fi
+        fi
+        local suffix=""
+        local function_name=$(echo "${BASH_REMATCH[1]}" | sed -E 's/(.*::)| //g')
+        # echo "// functioname ${function_name}"
+        MAP_FILE="map_list"
```

Comment:
This seems to come from somewhere else, maybe put it as a global variable or
function variable?